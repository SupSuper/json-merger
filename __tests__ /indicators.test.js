const jsonMerger = require("../dist/index");
const {testConfig} = require("./__helpers__/index");

describe("when merging two objects", function () {

    test("it should merge non-operation properties starting with the indicator prefix", function () {

        const object1 = {
            "a": {
                "aa": "original"
            }
        };

        const object2 = {
            "a": {
                "$$noOperation": true
            }
        };

        const result = jsonMerger.mergeObjects([object1, object2], testConfig());

        expect(result).toMatchSnapshot();
    });

    test("it should be possible to escape an indicator with an additional prefix character", function () {

        const object1 = {
            "a": {
                "aa": "original"
            }
        };

        const object2 = {
            "a": {
                "$$replace": {
                    "with": "not replaced"
                }
            }
        };

        const result = jsonMerger.mergeObjects([object1, object2], testConfig());

        expect(result).toMatchSnapshot();
    });

    test("it should be possible to escape an indicator but the value should be processed", function () {

        const object1 = {
            "a": {
                "aa": "original"
            }
        };

        const object2 = {
            "a": {
                "$$replace": {
                    "with": {
                        "$replace": {
                            "with": "not replaced"
                        }
                    }
                }
            }
        };

        const result = jsonMerger.mergeObjects([object1, object2], testConfig());

        expect(result).toMatchSnapshot();
    });
});
