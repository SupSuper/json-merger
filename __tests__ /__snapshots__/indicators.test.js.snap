// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when merging two objects it should be possible to escape an indicator but the value should be processed 1`] = `
"{
	\\"a\\": {
		\\"aa\\": \\"original\\",
		\\"$replace\\": {
			\\"with\\": \\"not replaced\\"
		}
	}
}"
`;

exports[`when merging two objects it should be possible to escape an indicator with an additional prefix character 1`] = `
"{
	\\"a\\": {
		\\"aa\\": \\"original\\",
		\\"$replace\\": {
			\\"with\\": \\"not replaced\\"
		}
	}
}"
`;

exports[`when merging two objects it should merge non-operation properties starting with the operation prefix 1`] = `
"{
	\\"a\\": {
		\\"aa\\": \\"original\\",
		\\"$$noOperation\\": true
	}
}"
`;
